### ISSUE TITLE
repeatable does not work if name includes brackets

### ISSUE THREAD

**Issue by kritsot** [Laravel-Backpack/community-forum#238]
# Bug report

### What I did

I used repeatable with a nested relationship property e.g. **relationship.property** and tried to add some rows

### What I expected to happen

I expected the rows to be saved

### What happened

The rows were not saved and after opening the developer tools console, a javascript error occured every time I pressed on the + button. 
_Uncaught Error: Syntax error, unrecognized expression: input[type=hidden][name=relationship[property]]_

### What I've already tried to fix it

The field_name should be escaped inside the repeatable.blade.php file.
A solution could be using jquery **$.escapeSelector**

### Is it a bug in the latest version of Backpack?

After I run ```composer update backpack/crud``` the bug... is it still there?

It is still there
### Backpack, Laravel, PHP, DB version

When I run ```php artisan backpack:version``` the output is:

### PHP VERSION:
PHP 8.1.6 (cli) (built: May 12 2022 23:30:39) (NTS)
Copyright (c) The PHP Group
Zend Engine v4.1.6, Copyright (c) Zend Technologies
    with Zend OPcache v8.1.6, Copyright (c), by Zend Technologies

### LARAVEL VERSION:
v9.42.2@607d7867c93706eae20e28e46679f8a66e2a23ec

### BACKPACK VERSION:
5.4.10@050b288caf2f60c79313f2b0b7d4fefd25177f3f

**Comment by kritsot** [comment]
Hello @tabacitu  and @pxpm .
After downloading and playing with the demo, I think this should be a priority as it doesn't work with some simple scenarios.
For example say I want to add a repeatable fields to **Ball** model named **repeatable_fields**.
*ball* is a morphOne relationship for Monster model.
Say I want to handle and save the ball relationship from MonsterCrudController as you currently do.

First there is the problem that as I think is true, you can't add it inside subfields of the ball field.
Since you handle subfields as repeatable, and you currently don't handle the case of a repeatable inside a repeatable.
But I think this should be a separate issue on its' own.

So I procced with converting the whole relationship to separate fields as I would do in version 4.

```
[
                'name'    => 'ball.name',
                'type'    => 'text',
                'label' => '',
                'wrapper' => [
                    'class' => 'form-group col-md-4',
                ]
            ],
            [
                'name'    => 'ball.country_id',
                'type'    => 'text',
                'label' => '',
                'entity'  => 'ball.country',
                'type'    => 'relationship',
                'wrapper' => [
                    'class' => 'form-group col-md-4',
                ]
            ],
            [
                'name'    => 'ball.repeatable_items',
                'type'    => 'repeatable',
                'label' => '',
                'subfields' => [
                    [
                        'name'    => 'name',
                        'type'    => 'text',
                        'label'   => 'Name',
                    ]
                ],
            ],
```

On loading I get a javascript error

**Uncaught Error: Syntax error, unrecognized expression: input[type=hidden][name=ball[repeatable_items]]**

This is solved by using *$.escapeSelector* in 3 lines 
wherever a jquery selector for *input[type=hidden][name=*  is used


Another issue I noticed is that if I set **repeatable_items** as translatable,
* no flag is shown
* the items are still saved inside the database as not translatable
* after editing the same page, no old values are rendered for repeatable_fields because it now expects the data to be translated. Eg wrapped inside the current locale **en**, which aren't. 

So with the same scenario, there are two extra issues
* cannot user repeatable inside subfields of a relationship
* nested fields(I don't know if this stands true, only for some types of relationship) do not support translatable repeatable.

Should I add them as separate issues and rewrite the same scenario?

**Comment by pxpm** [comment]
@kritsot amazing write-up, I am impressed with such detail 🥇 

About escaping the selectors, we already agreed to do it, and I will find time to make it happen during this week. 

If you played with demo and did some changes that will help me to realize the other issues you describe, it would **be awesome** if you can put that demo project up in github so that I can clone it and test your scenarios 👍 

Let me know,
Cheers

**Comment by kritsot** [comment]
@pxpm sure thing.

https://github.com/kritsot/backpack-demo
The branch is named **repeatable_bugs**

I have commented any reference or use related to dev tools, as I haven't purchased it.
MonsterCrudController.php line 834 and below describes the first issue.
I have commented the code causing trouble and replaced it with some other to be able to test the second issue.
Just remember to add **repeatable_items** to *$translatable* inside Ball.php to also be able to test that.

**Comment by pxpm** [comment]
I've merged and tagged a new PRO 2.0.17 version that has the fix for the css selectors (in some more places than those proposed by @kritsot at the time). 

I am moving the translatable issue as a separate issue and closing this one. 

Thanks for your patience and help here @kritsot 🙏
